applications/nrf5340_audio/src/audio/le_audio_rx.c

```diff
@@ -97,39 +82,44 @@ void le_audio_rx_data_handler(uint8_t const *const p_data, size_t data_size, boo
 
 	if (blocks_alloced_num >= CONFIG_BUF_BLE_RX_PACKET_NUM) {
 		/* FIFO buffer is full, swap out oldest frame for a new one */
-
-		void *stale_data;
+		struct audio_data *stale_audio_frame;
 		size_t stale_size;
 		num_overruns++;
 
 		if ((num_overruns % 100) == 1) {
 			LOG_WRN("BLE ISO RX overrun: Num: %d", num_overruns);
 		}
 
-		ret = data_fifo_pointer_last_filled_get(&ble_fifo_rx, &stale_data, &stale_size,
-							K_NO_WAIT);
+		ret = data_fifo_pointer_last_filled_get(&ble_fifo_rx, (void *)&stale_audio_frame,
+							&stale_size, K_NO_WAIT);
 		ERR_CHK(ret);
 
-		data_fifo_block_free(&ble_fifo_rx, stale_data);
+		struct net_buf *stale_buf = stale_audio_frame->data;
```

Comment:
Can this ever be NULL?

Reply:
It shouldn't, and there is an assert in the unref to catch NULL pointers.